src/alexapi/device_platforms/magicmirror.py

```diff
@@ -0,0 +1,172 @@
+from __future__ import print_function
+import time
+import os
+from abc import ABCMeta
+
+from baseplatform import BasePlatform
+import alexapi.bcolors as bcolors
+import BaseHTTPServer
+import threading
+import urllib2
+import urlparse
+
+"""
+Magic Mirror platform
+-----------------------------------------------------------------------------------------------------------------
+This device platform is made to integrate with the Magic Mirror project: https://github.com/MichMich/MagicMirror
+Specifically, this communicates with the MMM-AlexaPi module: https://github.com/dgonano/MMM-AlexaPi
+The two systems communicate over GET requests which,
+	- Update the Magic Mirror display with AlexaPi's status (No connection, idle, listening, processing, speaking)
+	- Allow the Magic Mirror to trigger a 'start listening' request
+"""
+
+class MagicmirrorPlatform(BasePlatform):
+	__metaclass__ = ABCMeta
+
+	def __init__(self, config):
```

Comment:
Do better ordering, blank lines, etc.
Something like 

```
+		if config['debug']:
 +			print("Initialising Magic Mirro platorm")

 +		super(MagicmirrorPlatform, self).__init__(config, 'magicmirror')

 +		self.HOST_NAME = self._pconfig['hostname']
 +		self.PORT_NUMBER = self._pconfig['port']
 +		self.MM_HOST = self._pconfig['mm_hostname']
 +		self.MM_PORT = self._pconfig['mm_port']
 +		self.HB_TIMER = self._pconfig['hb_timer']

 +		self.shutdown = False
 +		self.req_record = False
 +		self.httpd = ""
 +		self.serverthread = ""
```

don't mind the _+_.